tests/register/register.go

```diff
@@ -31,6 +38,97 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
-func Register(tType TestType, t types.Test) {
+func register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
 }
+
+// Registers t multiple times with different, compatible config versions
+func Register(tType TestType, t types.Test) {
+	// todo: how to handle errors? it errors on Empty Userdata, Preemption with default config, Preemption with base, default config, Preemption with no config but these test still need to run
+	var config typesInternal.Config
+	err := json.Unmarshal([]byte(t.Config), &config)
+	version, semverErr := semver.NewVersion(config.Ignition.Version)
```

Comment:
How should err and semverErr be handled? Tests like Empty Userdata, Preemption with default config, Preemption with base, default config, and Preemption with no config have purposeful errors and still need to be registered.

Reply:
What about the case where there is a really error that occurs? Or are you saying check for an error and if it is one of these tests, still register the test and return.